geotrek/core/models.py

```diff
@@ -244,6 +245,8 @@ def delete(self, *args, **kwargs):
         if not settings.TREKKING_TOPOLOGY_ENABLED:
             return super().delete(*args, **kwargs)
         topologies = list(self.topology_set.filter())
```

Comment:
```suggestion
        topologies = list(self.topology_set.filter())
```
hum.. this is not you current code, but why list(self.topology_set.filter()) ?
better perf with topology_exists = self.topology_set.exists() or .all().exists() isn't it ?